enhancements/olm/marketplace-api-deprecation.md

```diff
@@ -0,0 +1,130 @@
+
+If we determine that uninterrupted use of appregistry is desired or required, this proposal will need to include migration steps for all OperatorSource and CatalogSourceConfig APIs in-use in a cluster.
+
+Otherwise, a manual guide for migrating custom catalogs should be straightforward to follow.
+
+## Design Details
+
+### Graduation Criteria / Deprecation Plan
+
+Deprecation of the OperatorSource and CatalogSourceConfig APIs has been communicated in the [4.2 release notes](https://docs.openshift.com/container-platform/4.2/release_notes/ocp-4-2-release-notes.html)
+
+Removal of the marketplace operator will not yet remove the operator-marketplace namespace. This namespace is currently the designated namespace for global catalogs.
+
+### Upgrade / Downgrade Strategy
+
+On upgrade, the default catalog contents (redhat-operators, certified-operators, community-operators) will be converted into image-based CatalogSources that poll for updates.
+
+It should not be necessary to update any Subscriptions, as the new CatalogSources will have the same names as the old ones that were generated off of the OperatorSources.
+
+The OperatorHub config api may cease to be managed by the marketplace operator, and instead may be managed by OLM operator.
+
+### Version Skew Strategy
+
+During upgrade, there may be a period of time when the OperatorHub config API could be managed by two operators (marketplace and OLM).
+
+The new OLM version that is taking ownership of the OperatorHub config API should wait until the marketplace-operator is not available before reconciling the config API. This is also when OLM can perform the migration of the CatalogSources from "grpc/api" to "image/poll" types.
+
+This is only an issue when migrating ownership of the config api, which is a final step that is not required.
+
+#### Downgrades
+
+If OpenShift is downgraded, there are two cases that can occur
+
+ 1. defaults are **enabled**
+
+    - The marketplace operator will start up and recreate the default OperatorSources. This will adopt the CatalogSources of the same name, and overwrite their spec to point to the registry pod (which will be managed by marketplace).
+
+ 1. defaults are **disabled**
+
+    - The marketplace operator will start up, see no defaults, see no OperatorSources (because we are downgrading from a cluster that has removed them), and idle.
+    - The CatalogSources (migrated from OperatorSources on upgrade) will be using the existing image poll mechanism, which will continue to work in the previous version (it is in 4.4). They will continue to recieve updates, even though there is no OperatorSource, which may be confusing.
+    - The previous version of openshift will see operator content for version that was downgraded from.
```

Comment:
Will they be able to pull and use bundle images in 4.4?

Reply:
Yes